VirtoCommerce.JavaScriptShoppingCart.Data/VirtoCommerce.JavaScriptShoppingCart.Data.csproj

```diff
@@ -36,6 +36,9 @@
     <Reference Include="Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
       <HintPath>..\packages\Newtonsoft.Json.12.0.2\lib\net45\Newtonsoft.Json.dll</HintPath>
     </Reference>
+    <Reference Include="Omu.ValueInjecter, Version=2.3.3.0, Culture=neutral, PublicKeyToken=c7694541b0ac80e4, processorArchitecture=MSIL">
```

Comment:
Where do you need ValueInjecter?
In this stage we could add "manual" converters.
Later we want to use Automapper.

Reply:
https://github.com/VirtoCommerce/vc-module-javascript-shoppingcart/blob/dd62a2a9ba5743c1ee939ada7198cc6ec284108d/VirtoCommerce.JavaScriptShoppingCart.Data/Converters/CartConverter.cs#L601
In this case, the conversion comes from the object coming in the base class parameter. In fact, an object can have the type of an inherited class from the base PromotionReward. 
That is why this ValueInjecter is used. I think you can replace it with AutoMapper.